src/Microsoft.AspNetCore.Server.Kestrel/Filter/StreamSocketOutput.cs

```diff
@@ -49,9 +52,24 @@ public void Write(ArraySegment<byte> buffer, bool chunk)
 
         public Task WriteAsync(ArraySegment<byte> buffer, bool chunk, CancellationToken cancellationToken)
         {
-            // TODO: Use _outputStream.WriteAsync
-            Write(buffer, chunk);
-            return TaskUtilities.CompletedTask;
+            lock (_writeLock)
```

Comment:
The `_writeLock` was added to this class so the chunk prefix and suffix wouldn't get intermingled given concurrent/overlapping writes. The lock doesn't really work for async writes to a stream. I'm not sure how important of a concern this really is in the first place. It might be better just to remove the lock altogether.